src/extension.ts

```diff
@@ -282,33 +287,52 @@ export function activate(context: vscode.ExtensionContext): void {
     });
 
     initCommand(context, 'diagnostics.LogPoints.OpenScript', openScript);
+
+    initAsyncCommand(context, 'appService.showFile', async (node: IAzureNode<FileTreeItem>) => {
+        await fileEditor.showEditor(node);
+    });
+
+    initEvent(context, 'appService.fileEditor.onDidSaveTextDocument', vscode.workspace.onDidSaveTextDocument, (doc: vscode.TextDocument) => fileEditor.onDidSaveTextDocument(context.globalState, doc));
+
+
 }
 
+
 // tslint:disable-next-line:no-empty
 export function deactivate(): void {
 }
 
+function initEvent<T>(context: vscode.ExtensionContext, eventId: string, event: vscode.Event<T>, callback: (...args: any[]) => any) {
+
+    context.subscriptions.push(event(wrapAsyncCallback(eventId, (...args: any[]) => Promise.resolve(callback(...args)))));
+
+}
+
 // tslint:disable-next-line:no-any
 function initCommand(extensionContext: vscode.ExtensionContext, commandId: string, callback: (...args: any[]) => void): void {
-    // tslint:disable-next-line:no-any
+
     initAsyncCommand(extensionContext, commandId, async (...args: any[]) => callback(...args));
 }
 
 // tslint:disable-next-line:no-any
-function initAsyncCommand(extensionContext: vscode.ExtensionContext, commandId: string, callback: (...args: any[]) => Promise<void>): void {
-    // tslint:disable-next-line:no-any
-    extensionContext.subscriptions.push(vscode.commands.registerCommand(commandId, async (...args: any[]) => {
+function initAsyncCommand(context: vscode.ExtensionContext, commandId: string, callback: (...args: any[]) => Promise<any>) {
+    context.subscriptions.push(vscode.commands.registerCommand(commandId, wrapAsyncCallback(commandId, callback)));
+}
+
+
+function wrapAsyncCallback(commandId, callback: (...args: any[]) => Promise<any>): (...args: any[]) => Promise<any> {
+    return async (...args: any[]) => {
         const start = Date.now();
-        const properties: { [key: string]: string; } = {};
-        const output = util.getOutputChannel();
+        let properties: { [key: string]: string; } = {};
         properties.result = 'Succeeded';
-        let errorData: ErrorData | undefined;
+        let errorData: ErrorData | undefined = null;
+        const output = util.getOutputChannel();
 
         try {
             if (args.length === 0) {
```

Comment:
Remove this if/else case and just do `await callback(...args);`

(The only reason we had stuff like this was because we had initAsyncCommand<T> and <T> applied to the first argument. Like I said yesterday, the <T> stuff just complicated it with little benefit)